shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -785,7 +785,8 @@ fn build_slot_to_server(
     ensure!(frames.len() >= 2, "expected at least two fields");
 
     let ip = if let RedisFrame::BulkString(ref ip) = frames[0] {
-        String::from_utf8_lossy(ip.as_ref()).to_string()
+        std::str::from_utf8(ip.as_ref())
+            .map_err(|e| anyhow!("Failed to parse IP address as utf8 {e}"))?
```

Comment:
Would context be better? `std::str::from_utf8(ip.as_ref()).context("Failed to parse IP as utf8")?`

Reply:
Oh! thats very cool, I hadnt seen the `.context` method available on `Result` before.
Previously I had only seen the `.context` on the `Error`.